src/core/objective.jl

```diff
@@ -366,21 +367,22 @@ function _objective_min_fuel_cost_polynomial_nl(pm::AbstractPowerModel; report::
 
             cost_rev = reverse(gen["cost"])
             if length(cost_rev) == 1
-                gen_cost[(n,i)] = JuMP.@NLexpression(pm.model, cost_rev[1])
+                gen_cost[(n,i)] = JuMP.@expression(pm.model, cost_rev[1])
             elseif length(cost_rev) == 2
-                gen_cost[(n,i)] = JuMP.@NLexpression(pm.model, cost_rev[1] + cost_rev[2]*pg)
+                gen_cost[(n,i)] = JuMP.@expression(pm.model, cost_rev[1] + cost_rev[2]*pg)
             elseif length(cost_rev) == 3
-                gen_cost[(n,i)] = JuMP.@NLexpression(pm.model, cost_rev[1] + cost_rev[2]*pg + cost_rev[3]*pg^2)
+                gen_cost[(n,i)] = JuMP.@expression(pm.model, cost_rev[1] + cost_rev[2]*pg + cost_rev[3]*pg^2)
             elseif length(cost_rev) >= 4
                 cost_rev_nl = cost_rev[4:end]
-                gen_cost[(n,i)] = JuMP.@NLexpression(pm.model, cost_rev[1] + cost_rev[2]*pg + cost_rev[3]*pg^2 + sum( v*pg^(d+3) for (d,v) in enumerate(cost_rev_nl)) )
+                # TODO(odow): is this correct? Or should it be pg^(d+2.0)?
+                gen_cost[(n,i)] = JuMP.@expression(pm.model, cost_rev[1] + cost_rev[2]*pg + cost_rev[3]*pg^2 + sum( v*pg^(d+3.0) for (d,v) in enumerate(cost_rev_nl)) )

```

Comment:
This one is wrong again.

Reply:
Correctness aside, with this new system I would want to re-write this whole `if-then-else` block as one `JuMP.@expression(pm.model, sum(v*pg^(d-1) for (d,v) in enumerate(cost_rev)) )` is that going to work and produce the right JuMP data structure depending on the length of `cost_rev`?